library/core/src/ops/range.rs

```diff
@@ -688,13 +688,48 @@ impl<T> Bound<T> {
 
     /// Converts from `&mut Bound<T>` to `Bound<&T>`.
     #[inline]
+    #[unstable(feature = "bound_as_ref", issue = "80996")]
     pub fn as_mut(&mut self) -> Bound<&mut T> {
         match *self {
             Included(ref mut x) => Included(x),
             Excluded(ref mut x) => Excluded(x),
             Unbounded => Unbounded,
         }
     }
+
+    /// Maps a `Bound<T>` to a `Bound<U>` by applying a function to a contained value, returning a
+    /// Bound of the same kind
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(bound_map)]
+    /// use std::ops::Bound::*;
+    ///
+    /// let bound_string = Included("Hello, World!");
+    /// let bound_len = bound_string.map(|s| s.len());
+    ///
+    /// assert_eq!(bound_len, Included(13));
+    /// ```
+    ///
+    /// ```
+    /// #![feature(bound_map)]
+    /// use std::ops::Bound::*;
+    ///
+    /// let unbounded_string: Bound<String> = Unbounded;
+    /// let unbounded_len = unbounded_string.map(|s| s.len());
+    ///
+    /// assert_eq!(unbounded_len, Unbounded);
```

Comment:
```suggestion
    /// let bound = Included("Hello, World!");
    ///
    /// assert_eq!(bound.map(|s| s.len()), Included(13));
    /// ```
    ///
    /// ```
    /// #![feature(bound_map)]
    /// use std::ops::Bound::*;
    ///
    /// let unbounded: Bound<String> = Unbounded;
    ///
    /// assert_eq!(unbounded.map(|s| s.len()), Unbounded);
```

Looks a bit weird to have two version of each variable.

Reply:
sure! this is just what the example on `Option::map` did so I figured I'd mimic that as closely as possible